pkg/storage/bloom/v1/util.go

```diff
@@ -168,6 +168,10 @@ func (it *SliceIter[T]) At() T {
 	return it.xs[it.cur]
 }
 
+func (it *SliceIter[T]) Reset() {
+	it.cur = -1
```

Comment:
This `Reset()` has different semantics than the `Reset()` of the `BlockQuerier` ([here](https://github.com/grafana/loki/pull/11878/files#diff-ad299f9ea738f6f6d8e1a3b9fcc651506b3a241dbfbd95a60ab9d96b5b8b4d4aR139))

In the block querier, you can call `At()` after `Reset()` and you'll get the first item of the iterator. Here, you'll need to call `Next()` first, to do the same.

Reply:
You are right. I'm removing this one tho since it's a leftover from a previous impl that I was trying out

Reply:
Well, I'm actually leaving it, it's useful to test `PeekResetIterator`

Reply:
> In the block querier, you can call At() after Reset() and you'll get the first item of the iterator

Revisiting this, it's not correct. You are expected to call `Next` after `Reset` (or `Seek(0)`). Here's an test for that:
https://github.com/grafana/loki/blob/1748c29af710c310e24298be8e9398d172f2851d/pkg/storage/bloom/v1/builder_test.go#L276-L283

Anyway, in my last commit I removed this as it's no longer needed.